Zend/zend_ast.c

```diff
@@ -54,6 +54,18 @@ ZEND_API zend_ast * ZEND_FASTCALL zend_ast_create_znode(znode *node) {
 	return (zend_ast *) ast;
 }
 
+ZEND_API zend_ast * ZEND_FASTCALL zend_ast_create_fcc(void) {
+	zend_ast_fcc *ast;
+
+	ast = zend_ast_alloc(sizeof(zend_ast_fcc));
+	ast->kind = ZEND_AST_CALLABLE_CONVERT;
+	ast->attr = 0;
+	ast->lineno = ZEND_MAP_PTR_NEW_OFFSET();
+	ZEND_MAP_PTR_INIT(ast->fptr, NULL);
```

Comment:
I thought you could drop this field now that the map ptr offset is stored in `lineno`, which would allow you to drop this struct entirely, no? I don't even see `lineno` being used here for this purpose.

Reply:
> I don't even see `lineno` being used here for this purpose.

Whoops. `lineno` should actually be the `lineno` still:

I need to specially handle `ZEND_AST_CALLABLE_CONVERT` in `ext/opcache/zend_file_cache.c` to get a fresh “map ptr slot” when unserializing, because my understanding is that the slot used while serializing might already be taken when unserializing due to a different loading order.

And if I need to handle it specially, I might as well use a custom struct (similarly to `ZEND_AST_OP_ARRAY`).